Sources/NIOSSL/NIOSSLHandler.swift

```diff
@@ -732,27 +865,29 @@ extension NIOSSLHandler {
         // During the short writes we set the promise to `nil` to make sure they only arrive at the end.
         // Note that we make sure that there's always a single write, at the end, that holds the promise.
         while data.readableBytes > self.maxWriteSize, let slice = data.readSlice(length: self.maxWriteSize) {
-            bufferedWrites.append((data: slice, promise: nil))
+            bufferedActions.append(.write((data: slice, promise: nil)))
         }
 
         assert(data.readableBytes <= maxWriteSize)
-        bufferedWrites.append((data: data, promise: promise))
+        bufferedActions.append(.write((data: data, promise: promise)))
     }
 
     private func bufferFlush() {
-        bufferedWrites.mark()
+        bufferedActions.mark()
     }
 
-    private func discardBufferedWrites(reason: Error) {
-        while self.bufferedWrites.count > 0 {
-            let bufferedWrite = self.bufferedWrites.removeFirst()
-            bufferedWrite.promise?.fail(reason)
+    private func discardBufferedActions(reason: Error) {
+        while self.bufferedActions.count > 0 {
```

Comment:
This can be more concisely expressed as `while let bufferedAction = self.bufferedActions.popFirst()`.